query.go

```diff
@@ -302,6 +323,10 @@ func (q *query) run() {
 			return
 		}
 
+		// if the closet beta diverse peers have been queried, pick a non-diverse peer that's closer
+		// than any of the beta peer and add it to the set of peers to be queried.
+		q.allowClosestFilteredPeer()
+
```

Comment:
I think this approach of combining both the diverse and non-diverse lookups together is going to cause us problems/be confusing.

For example, by ending up with all the diverse and non-diverse peers in the same queue we have no way of getting out the results we want which are the K best diverse peers + the <=K best non-diverse peers.

One idea for how to handle this (I wouldn't be surprised if you found a better one) would be to:
- In the `runQuery` function create two separate query objects (not sure whether to use the same queryID here, but probably)
   - The first one has a filter and the rest of the parameters are normal. When peers are rejected for being non-diverse they are added
   - After the first one completes create a second one where the state is initialized based off of the first query
      - Going through the sorted list of peers from the peer queue (this function needs to be exported, or perhaps can be handled with an API similar to what we already have):
           - Find the K closest peers that are either Queried, Heard, or Rejected. Add the Queried and Heard ones (along with their referrers) to the peer queue
           - Add any Unreachable peers between up until the last of these peers to the peer queue
  - Run the second query
  - Combine the results into a single lookup result instance